server/routes/createAnagramicaRouter.js

```diff
@@ -22,11 +22,16 @@ const createAnagramicaRouter = config => {
     )
     .post((req, res) => {
       const { letters, words = [] } = req.body;
+      if (letters.length !== 10 && !/^[a-z]/.test(letters))
+        res.send({ error: 'Invalid letter selection' });
       const best = finder.best(toAlpha(letters));
-      const scores = words.reduce((total, rawWord) => {
-        const word = toAlpha(rawWord);
-        return Object.assign(total, { [word]: finder.find(word) });
-      }, {});
+      const scores =
+        Array.isArray(words) && words.length > 0
+          ? words.reduce((total, rawWord) => {
+              const word = toAlpha(rawWord);
+              return Object.assign(total, { [word]: finder.find(word) });
+            }, {})
+          : [];
```

Comment:
it's weird you return an empty array in on case and an object in another, should this default to an empty object?

(you presumably don't need the length check as reducing an empty array is a noop)

Reply:
good catch, should be an object 😞